tests/code-patches/2025-10-07-hb-balances-patch/index.test.mjs

```diff
@@ -0,0 +1,330 @@
+import fs from 'node:fs';
+import path from 'node:path';
+import { describe, it } from 'node:test';
+import assert from 'node:assert';
+import { assertNoResultError, createAosLoader } from '../../utils.mjs';
+import {
+  AO_LOADER_HANDLER_ENV,
+  DEFAULT_HANDLE_OPTIONS,
+  INITIAL_OPERATOR_STAKE,
+  STUB_ADDRESS,
+  STUB_BLOCK_HEIGHT,
+  STUB_HASH_CHAIN,
+  STUB_TIMESTAMP,
+} from '../../../tools/constants.mjs';
+
+const __dirname = path.dirname(new URL(import.meta.url).pathname);
+
+const patchFile = fs.readFileSync(
+  path.join(__dirname, '../../../patches/2025-10-07-hb-balances-patch.lua'),
+  { encoding: 'utf-8' },
+);
+
+const wasmMemory = fs.readFileSync(
+  path.join(
+    __dirname,
+    'qNvAoz0TgcH7DMg8BCVn8jF32QH5L6T29VjHxhHqqGE-2025-10-07-wasm-memory',
+  ),
+);
+
+const wasm = fs.readFileSync(
+  path.join(__dirname, 'CWxzoe4IoNpFHiykadZWphZtLWybDF8ocNi7gmK6zCg.wasm'),
+);
+
+const { handle: originalHandle, memory } = await createAosLoader({
+  wasm,
+  //  lua: patchFile,
+});
+const startMemory = memory;
+
+/**
```

Comment:
Might want to note why we're making another handle helper function here (i.e. needing to supply target in current fixture's module version).